infrastructure/settings_source_factory.go

```diff
@@ -199,6 +218,10 @@ func (s *SourceOptionsSlice) UnmarshalJSON(data []byte) error {
 				var o HTTPSourceOptions
 				err, opts = mapstruc.Decode(m, &o), o
 
+			case optType == "InstanceMetadata":
```

Comment:
let's update the test in https://github.com/cloudfoundry/bosh-agent/blob/master/app/config_test.go to make sure we dont lose this new configuration.